addons/website_sale_coupon/models/sale_order.py

```diff
@@ -8,6 +8,26 @@
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
+    def _try_pending_coupon(self):
+        if not request:
+            return
+
+        pending_coupon = request.session.get('pending_coupon', {})
+        status = {key: value for key, value in pending_coupon.items() if key == 'promo_code'}
+
+        if self and 'promo_code' in status:
+            coupon_code = status.get('promo_code')
```

Comment:
I don't think the `self` part is useful
can also get status before the if
Also name the variable the same as the key?
```suggestion
        promo_code = status.get('promo_code')
        if promo_code:
           
```

Reply:
The `apply_coupon` method needs an order for certain operations

Reply:
yes but when is `_try_pending_coupon` called without an order :thinking: ?

Reply:
For a visitor who didn't add anything yet to the cart?

Reply:
`_try_pending_coupon()` is only called by `recompute_coupon_lines()` which is supposed to be called on one or more sale order obviously.
Doesn't make sense to call `recompute_coupon_lines` on an empty saleorder recordset.
Either you have an order and you recompute/apply coupon, or you don't have an order and do nothing?

Reply:
I double checked, recompute does get called with an empty recordset when there is no website order present.

Reply:
yes with a `for order in self` inside it, which your overide is missing